plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/interceptor/SnapshottingToRegistryInterceptor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.machine.interceptor;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.ecr.AmazonECRClient;
+import com.amazonaws.services.ecr.model.CreateRepositoryRequest;
+import com.codenvy.auth.aws.ecr.AwsInitialAuthConfig;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.regex.Pattern;
+
+/**
+ * This interceptor creates repository in remote registry if it doesn't support repository creation on docker push command.
+ *
+ * @author Mykola Morhun
+ */
+@Singleton
+public class SnapshottingToRegistryInterceptor implements MethodInterceptor {
```

Comment:
@evoevodin , I used interceptor here because AWS ECR is specific registry. We should use AWS API to create repository before pushing and it is needed only for AWS.